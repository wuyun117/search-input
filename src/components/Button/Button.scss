@import "../styles/variables";

.Button {
  background: #fdc605;
  border: none;

  svg {
    height: 1.5rem;
    width: 1.5rem;
  }
}

.search__body--button {
  position: absolute;
  top: 0.1rem;
  right: 0.1rem;
  width: $search-box-height - 0.25rem;
  height: $search-box-height - 0.25rem;
}
