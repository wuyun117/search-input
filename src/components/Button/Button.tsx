import React, { MouseEvent } from "react";
import PropTypes from "prop-types";
import iconPath from "./icons.svg";
import "./Button.scss";

/**
 * <Button
 *   className="MyButton"
 *   onClick={() => console.log('Click')} 
 * />
 *
 * @prop {Function} onClick
 * @prop {mixed} ... All other props will be forwarded to the native DOM button.

 */

interface IButtonProps {
  className?: string;
  onClick: (event: MouseEvent) => void;
}

const Button: React.FC<IButtonProps> = (props: IButtonProps) => {
  const { className, onClick, ...otherProps } = props;

  return (
    <button
      type="button"
      className={"Button " + (className || "")}
      onClick={onClick}
      data-testid="search-alert-button"
      {...otherProps}
    >
      <svg viewBox="0 0 24 24" width="24" height="16">
        <use xlinkHref={iconPath + "#dls-icon-arrow-right"} />
      </svg>
    </button>
  );
}

Button.propTypes = {
  className: PropTypes.string,
  onClick: PropTypes.func.isRequired
}

export default Button
