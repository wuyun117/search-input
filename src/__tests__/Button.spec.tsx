import React from "react";
import { render, act, fireEvent } from "@testing-library/react";

import Button from "../components/Button/Button";

describe("Button component", () => {
  it("component render correctly", () => {
    const fn = jest.fn();
    const wrapper = render(<Button className="search__body--button" onClick={fn} />);
    expect(wrapper.container).toMatchSnapshot();
  });

  it("should have custom classname: search__body--button", async () => {
    const fn = jest.fn();
    const { getByTestId } = render(<Button className="search__body--button" onClick={fn} />);
    expect(getByTestId("search-alert-button")).toHaveClass("search__body--button");
  });

  it("should have only have classname Button", async () => {
    const fn = jest.fn();
    const { getByTestId } = render(<Button onClick={fn} />);
    expect(getByTestId("search-alert-button")).toHaveAttribute('class', 'Button ');
  });

  it("should trigger onclick event", async () => {
    const fn = jest.fn();
    const { getByTestId } = render(<Button className="search__body--button" onClick={fn} />);
    expect(fn).not.toBeCalled();
    await act(async () => {
      fireEvent.click(getByTestId("search-alert-button"));
    });
    expect(fn).toHaveBeenCalledTimes(1);
  });
});