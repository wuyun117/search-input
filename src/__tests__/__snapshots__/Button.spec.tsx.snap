// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component component render correctly 1`] = `
<div>
  <button
    class="Button search__body--button"
    data-testid="search-alert-button"
    type="button"
  >
    <svg
      height="16"
      viewBox="0 0 24 24"
      width="24"
    >
      <use
        xlink:href="icons.svg#dls-icon-arrow-right"
      />
    </svg>
  </button>
</div>
`;
